### PR TITLE
Implement -module flag in ch for running a script as a module

### PR DESCRIPTION
**Pull request by rhuanjl**
Fixes: #5274 

This PR adds a command line flag "-module" to ch which can be used to run a script as a module to avoid the need for wrapping module tests in an extra file of: WScript.LoadScriptfile(nameOfRealTest, "module");

I've added a simple test which runs an existing test file without it's wrapper to show that this works.

CC: @rwaldron

### CODE REVIEW

File: test/es6module/multiple-roots-circular.js [hunk 1]
```diff
@@ -5,8 +5,6 @@
 
 // Tests that circular overlapping module dependencies are all loaded before execution
 
-WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
```
Comment (rhuanjl) [inline test/es6module/multiple-roots-circular.js]: This change isn't really related - was reading over test cases to pick what to test the -module flag with and noticed this mistake from a previous commit of mine so removed it.

File: test/es6module/otherModule.js [hunk 2]
```diff
@@ -4,5 +4,5 @@
 //-------------------------------------------------------------------------------------------------------
 
 //test that dynamic import works from module
-//this file is imported only once - dynamicly from another module to check that the host is notified for that
```
Comment (rhuanjl) [inline test/es6module/otherModule.js]: As per above comment, this is an unrelated correction to one of my previous commits.

**Comment by sethbrenith** [comment]
[ChakraFull build](https://devdiv.visualstudio.com/DevDiv/Default/_build/results?buildId=1763509) based on commit de65f26b33ad0a53aa225f2314fc8b65a0eacff7 passed. I can't merge just yet due to some branch management shenanigans, but should be able to in a couple days. Thanks for the contribution!

**Comment by rhuanjl** [comment]
@sethbrenith thank you. Please let me know if you need this updated to latest or if any revision is required.

**Comment by rhuanjl** [comment]
@sethbrenith thanks for getting this merged

@rwaldron the -module flag is now available in ch in the master  branch of Chakreacore

**Comment by rhuanjl** [comment]
@sethbrenith did this just miss the cut for 1.10?

**Comment by sethbrenith** [comment]
Yes, master was locked down to bug fixes only for a couple of weeks, and once 1.10 was created, master reopened for feature checkins.

**Comment by dilijev** [comment]
Since this is a host-only change that might be helpful for folks using the future releases of 1.10, I think we can safely backport this change to 1.10 (the bar is not too high at the moment).

**Comment by rhuanjl** [comment]
@dilijev do you want/need me to do anything with regards to backporting this or will you handle it?

This should still merge cleanly with 1.10 at the moment.

**Comment by dilijev** [comment]
@rhuanjl I don't think there's anything. If we decide to backport it we will take care of creating and merging that PR.

**Comment by mathiasbynens** [comment]
FWIW, in `d8` the `*.mjs` extension implicitly enables this flag. Might be worth considering for `ch` too?

In other words, this:

```
ch foo.mjs
```

would be equivalent to:

```
ch -module foo.mjs
```

**Comment by zenparsing** [comment]
@mathiasbynens I think that's probably fine to add to `ch`.

On a broader level, we need to make sure that we are not giving JS developers the impression that the ".mjs" extension is necessary for modules. We probably shouldn't even be seen as recommending it, pending the outcome of the Node modules working group.
